core/src/main/scala/org/apache/spark/ui/ToolTips.scala

```diff
@@ -62,6 +62,13 @@ private[spark] object ToolTips {
     """Time that the executor spent paused for Java garbage collection while the task was
        running."""
 
+  val PEAK_EXECUTION_MEMORY =
+    """Execution memory refers to the memory used by internal data structures created during
+       shuffles, aggregations and joins when Tungsten is enabled. The value of this accumulator
+       should be approximately the sum of the peak sizes across all such data structures created
+       in this task. For SQL jobs, this only tracks all unsafe operators, broadcast joins, and
+       external sort."""
```

Comment:
I just commented on this elsewhere, but I have one concern about disabling this metric when Tungsten is disabled: users may run non-SQL/DataFrame jobs on a cluster where Tungsten has been enabled, so feature-flagging this based on whether Tungsten is on doesn't necessarily solve this confusion due to under-accounting of non-tracked memory.